modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaGetPortletGroupIdMethodCheck.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaGetPortletGroupIdMethodCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java") && !fileName.endsWith(".jsp") &&
+			!fileName.endsWith(".jspf") && !fileName.endsWith(".ftl")) {
+
+			return content;
+		}
+
+		Matcher getPortletGroupIdMatcher = _getPortletGroupIdPattern.matcher(
+			content);
+
+		while (getPortletGroupIdMatcher.find()) {
+			String methodCall = JavaSourceUtil.getMethodCall(
+				content, getPortletGroupIdMatcher.start());
+
+			if (fileName.endsWith(".java") &&
+				!hasClassOrVariableName("ThemeDisplay", content, methodCall)) {
+
+				return content;
+			}
+
+			content = StringUtil.replace(
+				content, methodCall,
+				StringUtil.replace(
+					methodCall, ".getPortletGroupId()", ".getScopeGroupId()"));
```

Comment:
I have some concerns about this. For the Java case, the check is fine becomes it checks if the method call is coming from a `ThemeDisplay` object. However, for the filetype cases, we are not checking that. It is possible that classes other than `ThemeDisplay` may have a `getPortletGroupId()`.

I do not know if there is a reliable way to check all cases, at least in a single check. We may want to create a dedicated check for each file type or just write this check for Java and maybe JSP.

Reply:
I understand your concerns. But, I tried to find 'getPortletGroupId()' method which is not ThemeDisplay Class and it was not found.

Another thing, themeDisplay is a variable available in '.jsp', '.jspf', and '.ftl'. Which makes the declaration in the class unnecessary. Therefore, for this variable themeDisplay cannot use 'hasClassOrVariableName' validation.

What do you think?

Reply:
> But, I tried to find 'getPortletGroupId()' method which is not ThemeDisplay Class and it was not found.

So, that may be true for our (Liferay) codebase, but not necessarily for a client codebase. When it comes to the client, we cannot be too sure.

>Another thing, themeDisplay is a variable available in `.jsp`, `.jspf`, and `.ftl`. Which makes the declaration in the class unnecessary.

But how do we know ___when___ the `ThemeDisplay` is available as a variable?